### ISSUE TITLE
Wrong architecture downloaded for gh-r on rpi-zero

### ISSUE THREAD

**Issue by jkp** [zdharma/zinit#476]
### Describe the bug
Wrong architecture is being detected, for downloading gh-r releases. A 64 bit binary is picked instead the 32bit arm binary.

### To Reproduce
On rpi-zero:

    zinit ice from"gh-r" as"program; zinit light sharkdp/bat

### Expected behaviour

`bat_0.17.1_armhf.deb` should be picked.

### Environment:
    echo $ZSH_VERSION $CPUTYPE $MACHTYPE
    5.7.1 armv6l arm

**Comment by pschmitt** [comment]
> bat_0.17.1_armhf.deb should be picked.

You mean `bat-v0.17.1-arm-unknown-linux-gnueabihf.tar.gz`, right?

**Comment by jkp** [comment]
Yeah - if that is that a binary release...was thrown off by the fact that it wasn't packaged as a deb....

**Comment by redxtech** [comment]
If you haven't solved this yet, you can use the `bpick` & `sbin` (from annex `z-a-bin-gem-node`) ices to pick the file to download from github releases, mark it as executable, and add it to path.

For this one, I would use this:
```zsh
zinit ice from"gh-r" as"program" sbin"bat-*/bat" bpick"*arm-unknown-linux-gnueabihf.*"
zinit light sharkdp/bat
```

I haven't tried it without the `sbin` ice, but I'm sure `bpick` should work the same without it. You can of course adjust the glob in `bpick` to match whichever file you want.

In this case, I would actually add another ice to add the completions:
```zsh
zinit ice from"gh-r" as"program" sbin"bat-*/bat" bpick"*arm-unknown-linux-gnueabihf.*" mv"bat-*/autocomplete/bat.zsh -> _bat"
```

**Comment by muuvmuuv** [comment]
Please add the "bpick" to the docs as a workaround. Using this resolved it on my iMac M1:

```
zinit ice from"gh-r" as"program" bpick"*arm*"
zinit light junegunn/fzf
```
